src/python/build_management/build_manager.py

```diff
@@ -864,8 +864,9 @@ class AndroidEmulatorBuild(RegularBuild):
 
   def setup(self):
     """Android Emulator build setup."""
+    self._pre_setup()
     emu_proc = android.emulator.EmulatorProcess()
-    emu_proc.create()
+    emu_proc.create(self.build_dir)
```

Comment:
We don't clean up build directory if revision does not change. Will this cause build directory to swell causing out of disk space?

Reply:
I think this always gets unpacked to the same place, so should not build up right @kalder  ?

Reply:
Yes, it is always unpacked to the same place and there already exists cleanup code (in emulator.py) to remove the archive immediately after unpacking it and the resulting emulator image and data during the EmulatorProcess's kill() method.